Initial version of python diff tool.

ptheywood opened the pull request:
Compares one .dat file to another, summarising the difference. Returns non zero code if any errors.
Optionally outputs differences as a CSV to simplify investigation

ptheywood committed:
Initial version of python diff tool.

Compares one .dat file to another, summarising the difference. Returns non zero code if any errors.
Optionally outputs differences as a CSV to simplify investigation

drewsilcock commented on tools/diff.py:
```diff
@@ -0,0 +1,163 @@
+        
+        # @todo - validate input file doesnt exist etc. 
+        with open(output, "w") as fp:
+            df.to_csv(output)
```
This opens the file pointer for file with name `output` but then writes to the file using the path name - could be `df.to_csv(fp)` instead?

ptheywood replied:
Yep that's what it was meant to be. looks like pandas' `to_csv` is fine with a string or a file pointer. Will fix tomorrow.

ptheywood committed:
fix and improve diff.py

Fix issue with --output, validate --output and introduce --force flag.
Additional generic improvements

ptheywood committed:
Implement elepson test for diff tool
